sentinel/test/mocha/lib/message-utils.js

```diff
@@ -0,0 +1,331 @@
+    });
+    it('tries to resolve complex values directly on the doc', () => {
+      utils._getRecipient({foo: {bar: {smang: 'baz'}}}, 'foo.bar.smang')
+        .should.equal('baz');
+    });
+    it('is doc.from if the recipient cannot be resolved', () => {
+      utils._getRecipient({from: 'foo'}, 'a-recipient')
+        .should.equal('foo');
+    });
+  });
+
+  describe('generate', () => {
+
+    it('adds uuid', done => {
+      sinon.stub(uuid, 'v4').returns('some-uuid');
+      const config = {};
+      const translate = null;
+      const doc = {};
+      const content = { message: 'xxx' };
+      const recipient = '+1234';
+      const messages = utils.generate(config, translate, doc, content, recipient);
+      expect(messages.length).to.equal(1);
+      const message = messages[0];
+      expect(message.message).to.equal('xxx');
+      expect(message.to).to.equal('+1234');
+      expect(message.uuid).to.equal('some-uuid');
+      done();
+    });
+
+    describe('truncation', () => {
+
+      it('does not truncate short sms', done => {
+        const sms = generateMessage(MAX_GSM_LENGTH);
+        const config = { multipart_sms_limit: 10 };
+        const translate = null;
+        const doc = {};
+        const content = { message: sms };
+        const recipient = '+1234';
+        const messages = utils.generate(config, translate, doc, content, recipient);
+        expect(messages.length).to.equal(1);
+        expect(messages[0].message).to.equal(sms);
+        expect(messages[0].original_message).to.equal(undefined);
+        done();
+      });
+
+      it('does not truncate short unicode sms', done => {
+        const sms = generateMessage(MAX_UNICODE_LENGTH, true);
+        const config = { multipart_sms_limit: 10 };
+        const translate = null;
+        const doc = {};
+        const content = { message: sms };
+        const recipient = '+1234';
+        const messages = utils.generate(config, translate, doc, content, recipient);
+        expect(messages.length).to.equal(1);
+        expect(messages[0].message).to.equal(sms);
+        expect(messages[0].original_message).to.equal(undefined);
+        done();
+      });
+
+      it('truncates long sms', done => {
+        const sms = generateMessage(1000);
+        const expected = sms.substr(0, 150) + '...';
+        const config = { multipart_sms_limit: 1 };
+        const translate = null;
+        const doc = {};
+        const content = { message: sms };
+        const recipient = '+1234';
+        const messages = utils.generate(config, translate, doc, content, recipient);
+        expect(messages.length).to.equal(1);
+        expect(messages[0].message).to.equal(expected);
+        expect(messages[0].original_message).to.equal(sms);
+        done();
+      });
+
+      it('truncates long unicode sms', done => {
+        const sms = generateMessage(1000, true);
+        const expected = sms.substr(0, 64) + '...';
+        const config = { multipart_sms_limit: 1 };
```

Comment:
as above - part limit of 1 means we can get 67 chars + `...` in the message.

Reply:
As per the above response.